src/inmanta/server/services/orchestrationservice.py

```diff
@@ -746,13 +747,19 @@ async def _put_version(
                     deleted_resource_sets=set(removed_resource_sets),
                     connection=connection,
                 )
-                resources_that_moved_resource_sets = rids_unchanged_resource_sets & set(rid_to_resource.keys())
+                resources_that_moved_resource_sets = rids_unchanged_resource_sets.keys() & set(rid_to_resource.keys())
```

Comment:
Why do we convert `rid_to_resource.keys()` to a set here?

Reply:
I think this can go